src/main/scala/services/Retry.scala

```diff
@@ -0,0 +1,25 @@
+package services
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.ExecutionContext.Implicits.global
+
+trait Retry {
+  def retryFuture[A](block: () => Future[A],
+                     acceptResult: A => Boolean,
```

Comment:
ie add `retryFailures: Seq[Exception]` which will define a sequence of exceptions that we want to retry.